### DISCUSSION TITLE
Mapping of varible names

### DISCUSSION THREAD

**Discussion by SmakshiA** [soot-oss/SootUp#999] [Q&A] [answered]
Is there any api to get mapping of jimple varible names generated from StmtGraph like l0,l1,l2 to the actual variable names i have defined?

**Comment by brenhub24** [comment, marked as answer]
Jimple is generated from the bytecode. If your bytecode doesn't have local variable names then there's no way Jimple has them.

See https://stackoverflow.com/questions/15357717/why-my-compiled-class-has-its-methods-local-variables-renamed

Maybe try to compile your bytecode with `-g` or `-g:vars`?

**Reply by swissiety** [reply]
precondition: you have to compile your code with `-g` like brenhub24 suggested.
beware:
1. its not possible to have all Locals that will be created named (i.e. those from the stack machine conversion)
2. currently there are cases (different variable of another block that can result in the same index position) where a wrong name is assigned - we are working on it.

**Reply by SmakshiA** [reply]
Thank you so much @brenhub24 and @swissiety, when compiled with -g command and we got the output.
